lib/pbench/server/indexing_tarballs.py

```diff
@@ -425,13 +441,33 @@ def sigquit_handler(*args):
 
                         if sigquit_interrupt[0]:
                             break
+                        if sighup_interrupt[0]:
+                            status, new_tb = self.collect_tb()
+                            if status == 0 and new_tb:
+                                # FIXME: add into deque
+                                # We have two choices to add into deque
+                                # Either sort the whole deque which might lead into starvation of
+                                # tarball or just extend the deque with new list
+
+                                # Here I will be using the second approach and we can discuss further
+                                tb_deque.extend(new_tb)
```

Comment:
Won't this add duplicates if we still haven't processed all the original tarballs?

Reply:
We currently sort them by size: there have been instances of large tarballs taking "forever" to index, so dealing with the small ones first helped a bit. Of course, sorting this batch doesn't do anything about subsequent tarballs and even with SIGHUP adding new tarballs and re-sorting, that will not be solved: if a large tarball starts being indexed, everybody else (including the SIGHUP handling) will have to wait. OTOH, given a steady supply of tarballs and a twitchy SIGHUP finger, it's possible that  a large tarball will be starved. But no matter what we do with the current implementation, this will continue to be a problem. 

So I think, we hold the SIGHUP handling up to (or down to) the same standard: sort the current set of tarballs and process them in that order. Scheduling them in parallel, possilby with different queues handling different size classes, is something to think about. But we all know that the heuristics break down in particular circurmstances.

Scheduling is hard (I have to add that to my sig...)

Reply:
The only thing that we might want to check for sanity's sake is that the new list is a superset of what is left of the old list. I'm not sure that's particularly important  and I don't think we should stop if that is not the case, but having an error message listing any "lost" tarballs might be worth having, even though it should never trigger.

Reply:
oh, I must have forgotten about replacing the list it just gets out of my mind. Checking for the superset and logging the lost tarballs is a good idea I will add that.